distribution/tools/plugin-cli/src/test/java/org/elasticsearch/plugins/cli/InstallPluginActionTests.java

```diff
@@ -404,6 +410,67 @@ public void testSomethingWorks() throws Exception {
         assertPlugin("fake", pluginDir, env.v2());
     }
 
+    public void testSlowSignatureVerificationMessage() throws Exception {
+        String url = "https://artifacts.elastic.co/downloads/elasticsearch-plugins/analysis-icu/analysis-icu-"
+            + Build.CURRENT.qualifiedVersion()
+            + ".zip";
+        final MessageDigest digest = MessageDigest.getInstance("SHA-512");
+
+        InstallPluginAction action = makeActionPluginThatDownloads(
+            "analysis-icu",
+            url,
+            null,
+            false,
+            ".sha512",
+            checksumAndFilename(digest, url),
+            newSecretKey(),
+            this::signature
+        );
+
+        final InstallPluginAction spied = spy(action);
+
+        // Control for timeout on waiting for signature verification to complete
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+
+        doAnswer(i -> {
+            i.callRealMethod();
+            countDownLatch.countDown();
+            return null;
+        }).when(spied).reportLongSignatureVerification();
+
+        // Make the slow verification acceptable delay artificially low for testing
+        doReturn(100L).when(spied).acceptableSignatureVerificationDelay();
+
+        doAnswer(i -> {
+            countDownLatch.await(); // wait until we trip the timer
+            i.callRealMethod();
+            return null;
+        }).when(spied).computeSignatureForDownloadedPlugin(any(InputStream.class), any(InputStream.class), any(PGPSignature.class));
+
+        installPlugin(new PluginDescriptor("analysis-icu", null), env.v1(), spied);
+        assertThat(terminal.getOutput(), containsString("The plugin installer is trying to verify the signature "));
+
+        // Clean-up the exiting plugin, let's try to reinstall with 'fast' random numbers
+        final List<PluginDescriptor> plugins = List.of(new PluginDescriptor("analysis-icu", null));
+        new RemovePluginAction(terminal, env.v2(), true).execute(plugins);
+
+        // Make the timer wait practically indefinitely, probably an overkill, but avoid flaky tests.
+        // Divide by two to meet the limitation of the Timer.schedule API.
+        doReturn(Long.MAX_VALUE / 2).when(spied).acceptableSignatureVerificationDelay();
+
+        // Make sure we don't see any slow error messages when the signature verification is fast
+        doCallRealMethod().when(spied)
+            .computeSignatureForDownloadedPlugin(any(InputStream.class), any(InputStream.class), any(PGPSignature.class));
+
+        int terminalLen = terminal.getOutput().length();
+
+        installPlugin(new PluginDescriptor("analysis-icu", null), env.v1(), spied);
+        assertThat(
+            terminal.getOutput().substring(terminalLen),
```

Comment:
Isn't this the same as just using `getOutput()` since terminalLen is the len of `getOutput()`?

Reply:
The new call of installPlugin will cause more messages to come out, so it will advance the length of the output from the length I capture in terminalLen. I'm  trying to cut out the previous install output which shows the error, and compare what the successful install produced in the output.

Reply:
MockTerminal has a `reset()` method to serve this purpose.

Reply:
Ah I didn't know, I'll use that instead, thanks!